### PR TITLE
ipython: 3.2 -> 4.0

### PR DESCRIPTION
**Pull request by FRidh**
This PR features [The Big Split](https://blog.jupyter.org/2015/08/12/first-release-of-jupyter/)

> IPython 3 was the last major monolithic release of IPython containing all of the language-agnostic code, such as the notebook, which is moving under the new name Jupyter. As part of that process, we have created several discrete packages for each component - nbformat for the notebook format, nbconvert for the notebook conversion machinery, notebook for the HTML notebook application, etc.

This PR includes all dependencies needed.
`ipython` is now a small package. If you want to have the notebook, you need to install
`pythonXXPackages.notebook`
and if you want to have the qtconsole. you need to install
`pythonXXPackages.qtconsole`

Guess I now need to rebase onto master because of the earlier update to `ipython 3.2.1`

cc maintainers @bjornfor @jgeerds

### CODE REVIEW

**Comment by FRidh** [comment]
Note that this PR might also have consequences for packages outside `python-packages.nix`.
For example, there is a haskell kernel, cc @peti

**Comment by pSub** [comment]
There are conflicts, can your resolve them please?

**Comment by FRidh** [comment]
@pSub rebased to master

**Comment by FRidh** [comment]
@geerds  gnome3.accerciser: init at 3.14.0 requires `ipythonLight` which I removed in this PR. Is the new `ipython` package sufficient? And if so, which version, 2.7 or 3.4?

**Comment by FRidh** [comment]
Built locally with `nix-shell -I nixpkgs=. -p XXPackages.notebook XXPackages.qtconsole` for all Python versions. Ran `jupyter notebook` and `jupyter qtconsole` successfully in all cases. I didn't use nox-review wip.

It would be nice to have this merged as soon as possible before anything breaks it again.

**Comment by FRidh** [comment]
The build failed due to running out of memory

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -2606,6 +2592,27 @@ let
...
+  pytestrunner = buildPythonPackage rec {
+    version = "2.6.2";
+    name = "pytest-runner-${version}";
+    
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/p/pytest-runner/${name}.tar.gz";
+      sha256 = "e775a40ee4a3a1d45018b199c44cc20bbe7f3df2dc8882f61465bb4141c78cdb";
+    };
+    
+    buildInputs = with self; [setuptools_scm pytest];
```
Comment (teh) [inline pkgs/top-level/python-packages.nix]: The nix style seems to be to have a space between `[` and content. I.e. `[ setuptools_scm pytest ]` (and in all other locations)

**Comment by teh** [comment]
I've tested this with nox-review which generated 94 outputs successfully and I've executed a few things (e.g. ipython) and they seem to work fine. IMO this is ready to go.

**Pull request merged**